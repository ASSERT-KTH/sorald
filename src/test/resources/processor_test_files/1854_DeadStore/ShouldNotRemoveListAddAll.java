/*
List.addAll() has a side effect, and so it should not be removed when identified as a dead store.
 */

public class ShouldNotRemoveListAddAll {
    public List<Integer> concatenate(List<Integer> lhs, List<Integer> rhs) {
        List<Integer> base = new ArrayList<>(lsh);
        boolean changed = base.addAll(rhs); // Noncompliant
        return base;
    }
}