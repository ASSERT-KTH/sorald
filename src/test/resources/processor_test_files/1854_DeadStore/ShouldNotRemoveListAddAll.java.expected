/*
List.addAll() has a side effect, and so it should not be removed when identified as a dead store.
 */

public class ShouldNotRemoveListAddAll {
    public List<Integer> concatenate(List<Integer> lhs, List<Integer> rhs) {
        List<Integer> base = new ArrayList<>(lsh);
        base.addAll(rhs);
        return base;
    }
}